cmake/modules/dts.cmake

```diff
@@ -135,6 +135,10 @@ endif()
 if(EXISTS ${DTS_SOURCE})
   # We found a devicetree. Check for a board revision overlay.
   if(DEFINED BOARD_REVISION)
+    zephyr_build_string(dts_board_string BOARD ${BOARD} BOARD_REVISION ${BOARD_REVISION})
+    if(EXISTS ${BOARD_DIR}/${dts_board_string}.overlay)
+      list(APPEND DTS_SOURCE ${BOARD_DIR}/${dts_board_string}.overlay)
+    endif()
```

Comment:
this doesn't seem to be a proper solution.

We should look at using `MERGE` flag on `zephyr_build_string(... MERGE)` :
https://github.com/zephyrproject-rtos/zephyr/blob/961b83a431224f57c698b7fe505075f6492536ee/cmake/modules/extensions.cmake#L1516-L1522

and then each returned filename must be checked for existence before appending to `DTS_SOURCE`.

Reply:
I first thought about using the MERGE option but using the raw result of this would mean also check for <board>.overlay and <board>_<identifier>.overlay

Reply:
true, and we don't want that in this case.

But the problem with existing code and this PR is that it doesn't catch the other SoC, CPU cluster, variants levels, so this PR only addresses the specific `<board>_<revision>.overlay` case.

Reply:
a revisioned board using a SoC with multiple CPU clusters would not pick up the SoC overlay.

For example: `-DBOARD=plank@5/soc1/cpuA` would in current proposal look up `plank_soc1_cpuA_5.overlay` and `plank_5.overlay`, but not `plank_soc1_5.overlay`.

Similar when using board variants.

Reply:
I have modified my commit so it looks for any combination of <identifier>. I think it is the least worse of doing it without having to modify the zephyr_build_string function or writing a new specialised function (but I'm not shure it would be better either). Please tell me what you think.